### PR TITLE
Renable email change confirmation tokens resets

### PR DESCRIPTION
**Pull request by airhorns**
This is on top of #819 but might need to be merged/rejected separately so I've broken it out. 

I think this functionality was accidentally disabled in 65b044d . When a user changes their email, an email is sent to the new email to ensure the user actually owns that new email and isn't sending emails to people that aren't them. A secret token is sent in that email and the user is only allowed to change their email if they can prove they know the secret that was only sent to that email.

This scheme breaks down if the secret is shared with multiple emails. Not that this is likely, but a user *could* change their email once to an email they control, and get a secret to confirm that email. If that
user then changed their email again to a target email they want to spam, and they knew that the same confirmation secret was used to confirm that second new email, they could craft and visit a link that confirmed it without ever having received the second confirmation email. That would allow them to spam the target email if the application sends lots of emails.

Long story short, the app used to this, and while its a bit farfetched
it may as well continue to do this.

This commit uncomments the line and adds a test.

### CODE REVIEW

**Comment by airhorns** [comment]
I might be missing some context here so feel free to close / ask for changes or what have you. Also, is this the kind of thing y'all would expect an integration test for?
